x/staking/client/cli/tx.go

```diff
@@ -1,26 +1,401 @@
 package cli
 
 import (
+	"bytes"
+	"encoding/hex"
+	"errors"
 	"fmt"
 
 	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
 
 	"github.com/cosmos/cosmos-sdk/client"
-	// "github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/client/tx"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/maticnetwork/bor/common"
+
+	// "github.com/maticnetwork/heimdall/bridge/setu/util"
+	"github.com/maticnetwork/heimdall/contracts/stakinginfo"
+	"github.com/maticnetwork/heimdall/helper"
+	hmTypes "github.com/maticnetwork/heimdall/types/common"
 	"github.com/maticnetwork/heimdall/x/staking/types"
 )
 
+var logger = helper.Logger.With("module", "staking/client/cli")
+
 // GetTxCmd returns the transaction commands for this module
 func GetTxCmd() *cobra.Command {
-	cmd := &cobra.Command{
+	stakingTxCmd := &cobra.Command{
 		Use:                        types.ModuleName,
 		Short:                      fmt.Sprintf("%s transactions subcommands", types.ModuleName),
 		DisableFlagParsing:         true,
 		SuggestionsMinimumDistance: 2,
 		RunE:                       client.ValidateCmd,
 	}
 
-	// this line is used by starport scaffolding # 1
+	stakingTxCmd.AddCommand(
+		ValidatorJoinTxCmd(),
+		StakeUpdateTxCmd(),
+		SignerUpdateTxCmd(),
+		ValidatorExitTxCmd(),
+	)
+
+	return stakingTxCmd
+}
+
+// ValidatorJoinTxCmd send validator join message
+func ValidatorJoinTxCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "validator-join",
+		Short: "Join Heimdall as a validator",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx := client.GetClientContextFromCmd(cmd)
+			clientCtx, err := client.ReadTxCommandFlags(clientCtx, cmd.Flags())
+			if err != nil {
+				return err
+			}
+
+			// get proposer
+			proposer := hmTypes.HexToAccAddress(viper.GetString(FlagProposerAddress))
+			if proposer.Empty() {
+				proposer = helper.GetFromAddress(clientCtx)
+			}
+
+			// get txHash
+			txhash := viper.GetString(FlagTxHash)
+			if txhash == "" {
+				return fmt.Errorf("transaction hash is required")
+			}
+
+			// get PubKey string
+			pubkeyStr := viper.GetString(FlagSignerPubkey)
+			if pubkeyStr == "" {
+				return fmt.Errorf("pubkey is required")
+			}
+
+			// convert PubKey to bytes
+			pubkeyBytes := common.FromHex(pubkeyStr)
+			if len(pubkeyBytes) != 65 {
```

Comment:
You can decompress pubkey -

```
ethcrypto "github.com/maticnetwork/bor/crypto"

pubk, err := ethcrypto.DecompressPubkey(key)
```

To compress it -- 

```
ethcrypto "github.com/maticnetwork/bor/crypto"

ethcrypto.CompressPubkey(&ecdsaPKey.PublicKey)
```

Reply:
`pubk, err := ethcrypto.DecompressPubkey(key)` return `ecdsaPKey.PublicKey` type which is not compatablie with hmCommon.PubKey.

Reply:
Just use `ethcryto.FromECDSAPub(pk)` to convert it to bytes.